core/lib/common/worker.js

```diff
@@ -343,6 +343,30 @@ module.exports = class Worker {
 		this.logger.log(`DUT has rebooted & is back online`);
 	}
 
+	/**
+	 * Wait for a service to be active/inactive
+	 *
+	 * @param {string} serviceName
+	 * @param {string} state
+	 * @param {string} target
+	 * @category helper
+	 */
+	async waitForServiceState(serviceName, state, target) {
+		return utils.waitUntil(async () => {
+			console.log(`waiting for ${serviceName} to be ${state}`);
+			return this.executeCommandInHostOS(
+				`systemctl is-active ${serviceName}`,
+				target,
+			).then((serviceStatus) => {
+				console.log(`serviceStatus: ${serviceStatus}`);
+				return Promise.resolve(serviceStatus === state);
+			}).catch((err) => {
+				Promise.reject(err);
+			});
+		}, false, 120, 250);
```

Comment:
Do we really need to check 120 times in a 250-millisecond delay? Just seems a little much. You might know while testing it out.

Reply:
@vipulgupta2048 This is trying for thirty seconds with a quarter second delay in between retries, instead of delaying thirty seconds between retries. This reduces the latency of detecting a service is running by a lot, and a quarter second delay is, in my opinion, pretty reasonable.